core/arch/x86/mangle.c

```diff
@@ -2403,7 +2405,17 @@ mangle_single_step(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr)
      * FIXME i#2144 : to be absolutely transparent, we should translate the
      * exception address as if we did not insert this nop.
      */
-    POST(ilist, instr, INSTR_CREATE_nop(dcontext));
+    instr_t* next = instr_get_next_app(instr);
+    /* Assumes popf cannot end a basic block. */
```

Comment:
This is not a valid assumption as blocks are truncated prior to a branch for various reasons.  Can we just have NULL be another condition.